pkg/cli/admin/release/extract_tools.go

```diff
@@ -320,20 +328,30 @@ func (o *ExtractOptions) extractCommand(command string) error {
 			missing.Insert(target.Mapping.Image)
 			continue
 		}
-		klog.V(2).Infof("Will extract %s from %s", target.Mapping.From, spec)
-		ref, err := imagereference.Parse(spec)
+		// Always try the registry/repo passed from the user first.  If image not found,
+		// proceed with the release info from the image spec
+		desired, err := imagereference.Parse(o.From)
 		if err != nil {
 			return err
 		}
-		target.Mapping.Image = spec
-		target.Mapping.ImageRef = imagesource.TypedImageReference{Ref: ref, Type: imagesource.DestinationRegistry}
-		if target.AsArchive {
-			willArchive = true
-			target.Mapping.Name = fmt.Sprintf(target.ArchiveFormat, releaseName)
-			target.Mapping.To = filepath.Join(dir, target.Mapping.Name)
-		} else {
-			target.Mapping.To = filepath.Join(dir, target.Command)
-			target.Mapping.Name = fmt.Sprintf("%s-%s", target.OS, target.Command)
+		digest, err := imagereference.GetSpecDigest(spec)
+		if err != nil {
+			return err
+		}
+		specFirst := fmt.Sprintf("%s@%s", desired.AsRepository(), digest)
+		klog.V(2).Infof("Will extract %s from %s", target.Mapping.From, specFirst)
+		target, willArchive, err = getValidTarget(specFirst, dir, releaseName, target, willArchive)
```

Comment:
I'd prefer passing `ref` here instead of that `specFirst`.

Reply:
the only way extract from user-provided image that was mirrored from a `registry.svc.ci.openshift.org/ocp/release:tag` that I've found is to 
1) substitute the registry/repo/name like so in specFirst
2) replace the `is.Tags.From.Name` for the extract target  in LoadReleaseInfo here: https://github.com/openshift/oc/blob/master/pkg/cli/admin/release/info.go#L778

 perhaps this needs to happen during the image mirror instead.

Reply:
resolving this, as I've updated to make all changes in pkg/cli/admin/release/info